src/main/java/com/iota/iri/service/milestone/impl/LatestMilestoneTrackerImpl.java

```diff
@@ -203,14 +198,19 @@ public MilestoneValidity analyzeMilestoneCandidate(Hash transactionHash) throws
      * {@link MilestoneSolidifier} that takes care of requesting the missing parts of the milestone bundle.<br />
      */
     @Override
-    public MilestoneValidity analyzeMilestoneCandidate(TransactionViewModel transaction) throws MilestoneException {
+    public boolean processMilestoneCandidate(TransactionViewModel transaction) throws MilestoneException {
         try {
             if (coordinatorAddress.equals(transaction.getAddressHash()) &&
                     transaction.getCurrentIndex() == 0) {
 
                 int milestoneIndex = milestoneService.getMilestoneIndex(transaction);
 
-                switch (milestoneService.validateMilestone(transaction, SpongeFactory.Mode.CURLP27, 1)) {
+                // if the milestone is older than our ledger start point: we already processed it in the past
+                if (milestoneIndex <= snapshotProvider.getInitialSnapshot().getIndex()) {
+                    return true;
+                }
+
+                switch (milestoneService.validateMilestone(transaction, milestoneIndex, SpongeFactory.Mode.CURLP27, 1)) {
```

Comment:
Let's try not to ignore codacy.

Even thought the way you did it is less verbose, I think it is still clearer and more readable to have the default case inside the switch brackets. You can also add a little comment:
`//We can consider the milestone candidate processed and move on w/o farther action`

Reply:
i think this is really unnecessary here since the switch is used as a faster / easier to read version of an if then else if then but if you think it helps i can add an empty default with that comment .. :P